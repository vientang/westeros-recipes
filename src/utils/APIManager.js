import superagent from 'superagent'

export default {
  get: (endpoint, params, callback) => {
    superagent
      .get(endpoint)
      .query(params)
      .set('Accept', 'application/json')
      .end((err, response) => {
        if (err) {
          callback(err, null)
          return err
        }
        callback(null, response)
      })
  }
}
