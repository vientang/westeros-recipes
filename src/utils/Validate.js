export default {
  email: (input) => {
    if (!input) return false
    const atSymbol = '@'
    const suffix = new RegExp(/\.[a-z]{2,4}\b/gi)
    return suffix.test(input) && input.includes(atSymbol)
  }
}
