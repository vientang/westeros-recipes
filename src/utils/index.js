import APIManager from './APIManager'
import Validate from './Validate'

export {
  APIManager,
  Validate
}
