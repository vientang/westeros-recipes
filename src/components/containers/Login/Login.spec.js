import React from 'react'
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { Login } from './index'
import { Validate } from '../../../utils'

test('Login snapshot test', () => {
  const component = shallow(<Login />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})

test('Login should validate if email string is acceptable', () => {
  const email = 'vt@vt.com'
  const validEmail = Validate.email(email) ? email : null
  const component = shallow(<Login />)
  if (validEmail) {
    component.setState({message: 'Your email is valid!'})
  } else {
    component.setState({message: 'Try again, your email is not valid'})
  }
  expect(component.state('message')).toEqual('Your email is valid!')
})

test('Login should clear validation message after 2000ms', () => {
  const component = shallow(<Login />)
  component.setState({valid: true, message: 'Your email is valid!'})
  expect(component.find('span').render().text()).toEqual('Your email is valid!')
  setTimeout(() => {
    expect(component.contains(<span key='' className='failed'> </span>)).toEqual(true)
  }, 2000)
})
