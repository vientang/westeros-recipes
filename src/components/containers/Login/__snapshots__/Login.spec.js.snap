// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot test 1`] = `
<section
  className="login-component"
>
  <form
    className="login-form"
  >
    <input
      className="form-input"
      id="email"
      onChange={[Function]}
      placeholder="Email"
      type="email"
    />
    <br />
    <input
      className="form-input"
      id="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
    <br />
    <button
      className="login-button"
      onClick={[Function]}
    >
      Log In
    </button>
  </form>
  <span
    className="failed"
  />
</section>
`;
